django/db/models/query.py

```diff
@@ -818,12 +818,33 @@ def _combinator_query(self, combinator, *other_qs, all=False):
         return clone
 
     def union(self, *other_qs, all=False):
+        # If the query is an EmptyQuerySet, replace the query with first
```

Comment:
I wonder if it would be simpler and more efficient to filter out all empty querysets. from self and other_qs, then use `qs[0].self._combinator_query('union', qs[1:], all=all)`, accounting for the case where qs might have zero or one elements.